File: pyjl/tests/yield_test.py
# @use_continuables
def generator_func():
    num = 1
    yield num
    num = 5
    yield num
    num = 10
    yield num

def generator_func_loop():
    num = 0
    for n in range(1, 10):
        yield num + n

def generator_func_loop_using_var():
    num = 0
    end = 12
    end = 16
    for n in range(1, end):
        yield num + n

class TestClass:
    def generator_func(self, ola: int):
        num = 1
        yield num
        num = 5
        yield num
        num = 10
        yield num

if __name__ == "__main__":
    for i in generator_func():
        print(i)
    print("-----------------------")
    for i in generator_func_loop():
        print(i)
    print("-----------------------")
    for i in generator_func_loop_using_var():
        print(i)
    print("-----------------------")
    testClass = TestClass()
    generator_func(testClass, 123) # Does not work
    for i in generator_func(testClass, 123):
        print(i)

File: pyjl/tests/yield_test.jl
using Continuables
using DataClass
function generator_func()
channel_generator_func = Channel(6)
num = 1
put!(channel_generator_func, num);
num = 5
put!(channel_generator_func, num);
num = 10
put!(channel_generator_func, num);
close(channel_generator_func)
channel_generator_func
end

function generator_func_loop()
channel_generator_func_loop = Channel(9)
num = 0
for n in (1:10 - 1)
put!(channel_generator_func_loop, (num + n));
end
close(channel_generator_func_loop)
channel_generator_func_loop
end

function generator_func_loop_using_var()
channel_generator_func_loop_using_var = Channel(15)
num = 0
end_ = 12
end_ = 16
for n in (1:end_ - 1)
put!(channel_generator_func_loop_using_var, (num + n));
end
close(channel_generator_func_loop_using_var)
channel_generator_func_loop_using_var
end

@dataclass mutable struct TestClass
end
@cont function generator_func(self::TestClass)
num = 123
cont(num);
num = 5
cont(num);
num = 10
cont(num);
end


function main()
for i in generator_func()
println(i);
end
println("-----------------------");
for i in generator_func_loop()
println(i);
end
println("-----------------------");
for i in generator_func_loop_using_var()
println(i);
end
println("-----------------------");
testClass::TestClass = TestClass()
for i in collect(generator_func(testClass))
println(i);
end
testClass::TestClass = TestClass()
funcs = [generator_func, generator_func_loop, generator_func_loop_using_var, testClass::generator_func]
for func in funcs
for i in func()
println(i);
end
end
end

main()
